### DISCUSSION TITLE
Best way to restore...

### DISCUSSION THREAD

**Discussion by luissimoes** [rdmtc/RedMatic#545] [Q&A]
I need to create a new SD Card with RaspberryMatic.... 
I have a backup of raspberryMatic.

I need then to "re-install" RedMatic"... but I wanted to put the configs I already have (basically Homekit)

According to the doc I made a copy of the files and the var dir --> i.e. I did a manual backup

Questions
- I do I do the restore?... by copying the secured files and dir over the newly installed?
- after t\his 
- -> can I keep the bridge I have in Home? (Hmekit)? or do I have to generate a new one (new QR Code)?
- 
Cheers

Luis

**Comment by blackflash73** [comment]
I had a similar problem, some time ago: needed to reformat my SD-Card. :/
Luckily I had a current RaspberryMatic Backup on an external USB drive. :)

1. Use SD-Card-Formatter to clean old flash drive (if applicable)
2. Create a new RaspberryMatic SD-card from Image (like once)
3. Do the initial Config or use Recovery Console
4. Restore from Backup using your Backup-File

In my case, all the AddOns and Data were restored, too (RedMatic, HAP-Homematic) :)

More details:
https://github.com/jens-maus/RaspberryMatic/wiki/Administration#backup--restore

**Reply by luissimoes** [reply]
Thanks for responding !

your point 3 is for RaspberryMatic? Or Redmatic? 
because I am not sure where the console is ….

but on that case you had a “full backup” of the Redmatic? 
so you didby need to copy the files and dirs as mentioned in the Redmatic documentation

thanks!
Luis

**Reply by blackflash73** [reply]
Yes, if you use a fresh RaspberryMatic SD-Card it needs some initial configuration. After that you can restore from backup in the RaspberryMatic control panel. 

If the RaspberryMatic hangs during boot, you can use the button on the RPI-RF-MOD radio module to enter the RaspberryMatic Recovery Console (WebUI) which is a minimal fail safe mode. There you have an option to recover from Backup.

**Reply by blackflash73** [reply]
If you plug in a formatted USB-Thumbdrive to the RPi running RaspberryMatic, you can enable (daily) automatic backups to this drive. That's what I did. 

Since lately, you have an option to create (and download) a Backup before updating RaspberryMatic. Good point if sth. goes wrong during update (power outage...)

Or you can create manual backups from the RaspberryMatic control panel.

**Reply by luissimoes** [reply]
🙏🏾
thanks!

**Comment by hobbyquaker** [comment]
Beschrieben unter https://github.com/rdmtc/RedMatic/wiki/Backup: RedMatic nach dem Restore neu installieren, Flows/Credentials/var liegen im CCU-Backup. Geschlossen.

---
*Written by Claude Fable on behalf of hobbyquaker.*
